src/k8s/request.js

```diff
@@ -238,10 +287,101 @@ const addCloudInit = (vm, basicSettings) => {
   }
 };
 
-const addDisk = (vm, diskSpec) => {
+const addStorages = (vm, storages, networks) => {
+  if (storages) {
+    for (const storage of storages) {
+      if (storage.attachStorage) {
+        addPersistentVolumeClaimVolume(vm, networks, storage);
+      } else {
+        addDataVolume(vm, networks, storage);
+      }
+    }
+  }
+};
+
+const addDataVolume = (vm, networks, volume) => {
+  addDataVolumeTemplate({
+    metadata: {
+      name: volume.name
+    },
+    spec: {
+      accessModes: ['ReadWriteOnce'],
+      pvc: {
+        resources: {
+          requests: {
+            storage: `${volume.size}Gi`
+          }
+        },
+        storageClassName: volume.storageClass
+      },
+      source: {}
+    },
+    status: {}
+  });
+
+  addVolume(vm, {
+    name: volume.name,
+    dataVolume: {
+      name: volume.name
+    }
+  });
+
+  addBootableDisk(
+    vm,
+    networks,
+    {
+      name: volume.name,
+      volumeName: volume.name,
+      disk: {
+        bus: VIRTIO_BUS
+      }
+    },
+    volume.isBootable
+  );
+};
+
+const addPersistentVolumeClaimVolume = (vm, networks, volume) => {
+  const { name } = volume.attachStorage.metadata;
+
+  addVolume(vm, {
+    name,
+    persistentVolumeClaim: {
+      claimName: name
+    }
+  });
+
+  addBootableDisk(
+    vm,
+    networks,
+    {
+      name,
+      volumeName: name,
+      disk: {
+        bus: VIRTIO_BUS
```

Comment:
disk spec should be taken from template's default disk. See `ANNOTATION_DEFAULT_DISK`. If it exists, than you should use what is defined there (bus and possibly other stuff) and change only name, volumename and boot order. If the annotation is not there (for example in case of template from import-vm-apb) we could take the first found disk. If no disk is found than we could fallback to this.

Reply:
This can be done as followup